src/lib/Libifl/int_status.c

```diff
@@ -81,7 +205,471 @@ PBSD_status(int c, int function, char *objid, struct attrl *attrib, char *extend
+ * @param[in,out] head_b - reference to head element of list contains b. Reference is updated if prev_b is null
+ * 
+ * @return void
+ */
+static void
+append_bs(struct batch_status *a, struct batch_status *b, struct batch_status *prev_b, struct batch_status *head_a, struct batch_status **head_b)
+{
+	if (!a) {
+		for (a = head_a; a->next; a = a->next)
+			;
+		a->next = b;
+		if (prev_b) {
+			prev_b->next = b->next;
+		} else {
+			*head_b = b->next;
+		}
+		b->next = NULL;
+	}
+}
+
+/**
+ * @brief
+ * aggregate_queue:
+ *	aggregate two queues reported by two server instances
+ * @param[in,out] sv1 - server1 list. This will be also the final aggregated list.
+ * @param[in,out] sv2 - Server2 list. Passing reference as the first element can be moved to sv1
+ * 
+ * @return void
+ */
+static void
+aggregate_queue(struct batch_status *sv1, struct batch_status **sv2)
+{
+	struct batch_status *a = NULL;
+	struct batch_status *b = NULL;
+	struct batch_status *prev_b = NULL;
+
+	for (b = *sv2; b; prev_b = b, b = b->next) {
+		for (a = sv1; a; a = a->next) {
+			if (a->name && b->name && !strcmp(a->name, b->name)) {
+				aggr_job_ct(a, b);
+				aggr_resc_ct(a, b);
+				break;
+			}
+		}
+
+		if (!a)
+			append_bs(a, b, prev_b, sv1, sv2);
+	}
+}
+
+/**
+ * @brief
+ * aggregate_svr:
+ *	aggregate two servers reported by two server instances
+ * @param[in,out] sv1 - server1 list. This will be also the final aggregated list.
+ * @param[in] sv2 - Server2 list.
+ * 
+ * @return void
+ */
+static void
+aggregate_svr(struct batch_status *sv1, struct batch_status *sv2)
+{
+	aggr_job_ct(sv1, sv2);
+	aggr_resc_ct(sv1, sv2);
+}
+
+/**
+ * @brief
+ *	wrapper function for PBSD_status
+ *	gets aggregated value for all servers.
+ *
+ * @param[in] c - communication handle
+ * @param[in] id - job id
+ * @param[in] attrib - pointer to attribute list
+ * @param[in] extend - extend string for req
+ * @param[in] cmd - command
+ *
+ * @return	structure handle
+ * @retval	pointer to batch_status struct		success
+ * @retval	NULL					error
+ *
+ */
+struct batch_status *
+PBSD_status_aggregate(int c, int cmd, char *id, void *attrib, char *extend, int parent_object, struct attrl *rattrib)
+{
+	int i;
+	int rc = 0;
+	struct batch_status *ret = NULL;
+	struct batch_status *next = NULL;
+	struct batch_status *cur = NULL;
+	svr_conn_t *svr_connections = get_conn_svr_instances(c);
+	int num_cfg_svrs = get_num_servers();
+	int *failed_conn;
+	char server_out[PBS_MAXSERVERNAME + 1];
+	char server_name[PBS_MAXSERVERNAME + 1];
+	int single_itr = 0;
+	char job_id_out[PBS_MAXCLTJOBID];
+	uint server_port;
+	int start = 0;
+	int ct;
+	struct batch_status *last = NULL;
+
+	if (!svr_connections)
+		return NULL;
+
+	failed_conn = calloc(num_cfg_svrs, sizeof(int));
+
+	if (pbs_client_thread_init_thread_context() != 0)
+		return NULL;
+
+	if (pbs_verify_attributes(random_srv_conn(svr_connections), cmd, parent_object, MGR_CMD_NONE, (struct attropl *) attrib) != 0)
```

Comment:
Here also it is better to use pbs_verify_attributes_wrapper() so that we can catch PBSE_NOSERVER early if it happens.

Reply:
It does not matter as status API's has to give partial output.

Reply:
Yes but what if the server that pbs_verify_attributs() chooses is down, we still need to talk to another server right ?

Reply:
That is handled today within random_srv_conn